src/asm.h

```diff
@@ -2,18 +2,18 @@
 
 #include "types.h"
 
-typedef struct {
-    u32 hardware_id;
-    u16 hardware_version;
+struct HardwareInfo {
+    u32 id;
+    u16 version;
     u32 manufacturer;
-} hardware_infos;
+};
```

Comment:
while we are at it, we could add a
```c
union {
    u32 id;
    struct {
        u16 id_low;
        u16 id_high;
    }
}
```
I checked, it works and it would be more efficient (no need to deal with shr) while keeping the easy-to-use u32. We have to pay attention to en endianess, though.